src/NuGet.Core/NuGet.ProjectManagement/Projects/FolderNuGetProject.cs

```diff
@@ -109,12 +103,12 @@ public override Task<IEnumerable<PackageReference>> GetInstalledPackagesAsync(Ca
                     await PackageExtractor.ExtractPackageAsync(
                         downloadResourceResult.PackageStream,
                         PackagePathResolver,
-                        nuGetProjectContext.PackageExtractionContext,
-                        PackageSaveMode,
+                        nuGetProjectContext.PackageExtractionContext ?? new PackageExtractionContext(),
                         token));
             }
 
-            if (PackageSaveMode.HasFlag(PackageSaveModes.Nupkg))
+            var packageSaveMode = GetPackageSaveMode(nuGetProjectContext);
+            if (packageSaveMode.HasFlag(PackageSaveModes.Nupkg))
```

Comment:
I know this is not your change, but I'd like to understand why we use flags enum here, where this is representing a mode